pkg/dns/dns.go

```diff
@@ -25,66 +25,41 @@ import (
 	"github.com/pkg/errors"
 )
 
-type DNSProvider int
-
 const (
-	DNSNone DNSProvider = iota
-	DNSManual
-	DNSRoute53
+	// DNSManual represents manual DNS configuration.
+	DNSManual = "manual"
+	// DNSRoute53 represents DNS managed in Route 53.
+	DNSRoute53 = "route53"
 )
 
-type route53Provider struct {
-	ZoneID       string `hcl:"zone_id,optional"`
-	AWSCredsPath string `hcl:"aws_creds_path,optional"`
-}
-
-type dnsProvider struct {
-	Route53 *route53Provider `hcl:"route53,block"`
-	Manual  *manualProvider  `hcl:"manual,block"`
-}
-
-type manualProvider struct{}
-
-type Config struct {
```

Comment:
>This allows to do nicer naming of functions

In what way does keeping `dns.Config` help having nicer function naming? I don't follow.

What is `config.Dns{}`? Did you mean `dns.Config` here, too?

>This way, you can also avoid changing the interface between the dns package and platform package

How did we change the interface between these packages? I did rename `ParseDNS` to `Validate` (because the function validates rather than parses), however the old function was also called "statically", i.e. it wasn't a method.

>This also makes simpler and more consistent handling of DNS paramaters

Parameters such as...?

I've removed `dns.Config` because IMO it makes things simpler, and simpler is good. I understand you're suggesting we keep `dns.Config{}`, however I don't understand what fields - if any - you want to have in this type. I'm also not fond of the name `dns.Config`. Could you elaborate?

Reply:
>>This allows to do nicer naming of functions
>In what way does keeping dns.Config help having nicer function naming? I don't follow.

It allows you to build functions with context, so instead of having:
```go
package dns

func ValidateProviderName(name string) error {}
````
, you can have:
```go
package dns

type Config struct {
  Provider string
}

func  (c *Config) Validate() error { ... }
```

It's shorter name, less parameters to pass and IMO more pleasant to work with, as then as the caller, you can use it like `c.DNSConfig.Validate()`, instead of `dns.ValidateProviderName(c.DNSProviderName)`.

>What is config.Dns{}? Did you mean dns.Config here, too?

Yes, sorry about that.

>>This way, you can also avoid changing the interface between the dns package and platform package

>How did we change the interface between these packages? I did rename ParseDNS to Validate (because the function validates rather than parses), however the old function was also called "statically", i.e. it wasn't a method.

You changed it, as you don't use `dns.Config` to compose your configuration anymore, but 2 separate fields. Which we will have to copy-paste to other providers, if we want to provide the matrix of DNS settings.

>>This also makes simpler and more consistent handling of DNS paramaters

>Parameters such as...?

Parameters you have separately now and you suggested yourself, so `Provider` and `Zone`.

>I'm also not fond of the name dns.Config. Could you elaborate?

Not sure what would be the better name here. We deal with Lokomotive DNS configuration after all, so `github.com/kinvolk/lokomotive/pkg/dns.Config` seems reasonable to me.

Reply:
I've reintroduced `dns.Config` as a block - see https://github.com/kinvolk/lokomotive/pull/422#discussion_r425391553.